src/summarizedexperiment/_combiners.py

```diff
@@ -0,0 +1,175 @@
+from typing import Optional, Sequence, Tuple, Union, Literal, MutableMapping
+from functools import reduce
+import pandas as pd
+import numpy as np
+import scipy.sparse as sp
+
+ArrayTypes = Union[np.ndarray, sp.lil_matrix]
+
+from ._validators import validate_names, validate_shapes
+from .dispatchers.combiners import combine, combine_prefer_left, combine_ignore_names
+
+
+def _impose_common_precision(x: ArrayTypes, y: ArrayTypes):
+    """Ensure input arrays have compatible dtypes.
+
+    Args:
+        x (ArrayTypes): first array.
+        y (ArrayTypes): second array.
+    """
+    dtype = np.find_common_type([x.dtype, y.dtype], [])
+    if x.dtype != dtype:
+        x = x.astype(dtype)
+    if y.dtype != dtype:
+        y = y.astype(dtype)
+
+
+def combine_metadata(ses: Sequence["BaseSE"]) -> Optional[MutableMapping]:
+    """Combine metadata across experiments.
+
+    Args:
+        ses (Sequence[BaseSE]): "SummarizedExperiment" objects.
+
+    Returns:
+        combined_metadata (Optional[MutableMapping]): combined metadata.
+    """
+    combined_metadata = []
+    for se in ses:
+        if se.metadata:
+            combined_metadata.extend(se.metadata.values())
+    return dict(enumerate(combined_metadata))
+
+
+def combine_concatenation_axis(
+    ses: Sequence["BaseSE"], experiment_metadata: Literal["rowData", "colData"]
```

Comment:
I find the use of `experiment_metadata` confusing. May be rename this to `experiment_attribute`? This applies to every usage of this word

Reply:
also you probably need a check if someone forces their way into using something other than `rowData` or `colData`

Reply:
it's not possible to have something other than `rowData` and `colData`. i hardcode them under the hood, so the user doesn't have any control over these parameters... not sure it makes sense to validate this?

Reply:
I'm thinking of this from downstream SE-derivatives that might need modify this behavior like SCE.